core/src/crypto/BLAKE.cpp

```diff
@@ -55,11 +59,32 @@ namespace ledger {
         }
 
         std::vector<uint8_t> BLAKE::blake2b(const std::vector<uint8_t>& data, size_t outLength, size_t offset) {
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
             std::vector<uint8_t> hash(outLength);
-            BLAKE2B_CTX blakeState;
             BLAKE2b_Init_default(&blakeState, outLength);
             BLAKE2b_Update(&blakeState, data.data() + offset, (data.size() - offset));
             BLAKE2b_Final(hash.data(), &blakeState);
+#else
+            // The OpenSSL official documentation (and developers too) recommend to use the
+            // EVP interface. The code for < 1.1 above was required because OpenSSL didn’t have
+            // direct support of it
+            EVP_MD_CTX * mdctx = EVP_MD_CTX_new();
```

Comment:
Is this resource allocation, if yes were should we free it?
I know this is urgent, so put TODO to wrap it to smart pointer to have automatic resources handling.

Reply:
Yeah I’ll fix that in the PR too, it should be removed by the end of the function. We’ll need to find a better scheme later not to do those dynamic allocation on the fly, as they might be costly.

Reply:
Yeap the same staff I tried to merge in https://github.com/LedgerHQ/lib-ledger-core/pull/307